lib/request.js

```diff
@@ -0,0 +1,120 @@
+      n[key] = parseBuffers(value[key]);
+      return n;
+    }, {});
+  } else {
+    return value;
+  }
+};
+
+function followRedirect(cfg, response, callback) {
+  if (cfg.maxRedirects < 1) {
+    throw new Error('Too many redirects');
+  }
+
+  cfg.maxRedirects -= 1;
+
+  let parsedUrl = response.headers.location && url.parse(response.headers.location);
+
+  Object.assign(cfg, {host: parsedUrl.host, path: parsedUrl.pathname, port: parsedUrl.protocol.includes('https') ? 443 : 80});
+
+  return request(cfg, 'GET', '', {}, Buffer.from(JSON.stringify({})), (err, res) => {
+    if (err) {
+      return callback(err);
+    }
+    return callback(null, res);
+  });
+}
+
+function request(cfg, method, pathname, headers, body, callback) {
+  cfg.token && (headers['Authorization'] = `Bearer ${cfg.token}`);
+  cfg.keys && (headers['X-Authorization-Keys'] = JSON.stringify(cfg.keys));
+  cfg.webhook && (headers['X-Webhook'] = cfg.webhook); // TODO: Deprecate
+  cfg.convert && (headers['X-Convert-Strings'] = 'true');
+  headers['X-Stdlib-Stack-Trace'] = process.env.__STDLIB_STACK_TRACE || '';
+  cfg.bg && (pathname += `:bg${typeof cfg.bg === 'string' ? '=' + encodeURIComponent(cfg.bg) : ''}`);
+
+  let responded = false;
+
+  let req = (cfg.port === 443 ? https : http).request({
+    host: cfg.host,
+    method: method,
+    headers: headers,
+    port: cfg.port,
+    path: `${cfg.path}${pathname}`,
+    agent: false
+  }, function (res) {
+
+    let buffers = [];
+
+    res.on('data', chunk => buffers.push(chunk));
+    res.on('end', () => {
+      var response = Buffer.concat(buffers);
+      var contentType = (res.headers['content-type'] || '').split(';')[0];
+
+      if (contentType === 'application/json') {
+        response = response.toString();
+        try {
+          response = JSON.parse(response);
+        } catch(e) {
+          response = null;
+        }
+      } else if (contentType === 'text/plain') {
+        response = response.toString();
+      }
+
+      response = parseBuffers(response, contentType);
+      responded = true;
+
+      if (((res.statusCode / 100) | 0) !== 2 && ((res.statusCode / 100) | 0) !== 3) {
+        let message = typeof response === 'object' ?
+          (response && response.error && response.error.message) || ('Unspecified error running remote Standard Library function "' + names.join('.') + '"') :
+          response;
+          let error = new Error(message);
+          if (response.error && response.error.hasOwnProperty('details')) {
+            error.details = response.error.details;
+          }
+          if (response.error && response.error.hasOwnProperty('type')) {
+            error.type = response.error.type;
+          }
+          if (response.error && response.error.hasOwnProperty('stack')) {
+            error.stack = response.error.stack;
+          }
+          return callback(error, response);
+      } else if (((res.statusCode / 100) | 0) === 3) {
```

Comment:
I would actually set this to only follow 301 or 302 here, other codes have other behavior: https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#3xx_Redirection